pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -153,6 +154,13 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environme
 	for _, service := range push.Services(proxy) {
 		destRule := push.DestinationRule(proxy, service)
 		for _, port := range service.Ports {
+			var mtlsPolicy *authn.Policy
+			if features.UseAutoPilotMTLS.Get() {
+				policyCfg := env.IstioConfigStore.AuthenticationPolicyForWorkload(service, nil, port)
```

Comment:
Please donnt run this env.IstioConfigStore query here. Its going to negate all the benefits of doing the Sidecar, the caching, pre processing etc. The entire thing needs to be precomputed [whether the service requires strict or not] and stored in the pushContext. Otherwise, with this code, turning on auto mtls will result in a super slow Istio which will cause people to turn this thing off.